python/lsst/obs/ctio0m9/ingest.py

```diff
@@ -58,6 +58,35 @@ def translate_visit(self, md):
         """
         return mjdToVisit(md.get("DATE-OBS"))
 
+    def translate_imgType(self, md):
+        """Get the image type (e.g. bias, dark, flat etc) from the header
+        Translator function derived from a very small dataset from the observatory
+        i.e. may well need adding to when new string values are found
+        """
+        val = md.get("IMAGETYP").rstrip().lstrip()
+        conversion = {'dflat': 'flat', 'DOME FLAT': 'flat',
+                      'zero': 'bias',
+                      'object': 'object'}
+        if val in conversion:
+            return conversion[val]
+        else:
+            return 'NOT_TRANSLATED'
+
+    def translate_wavelength(self, md):
+        """Get the monochromator wavelength from the header, for flats only.
+        Method will need ammending if/when this value is written elsewhere
+        """
+        val = md.get("OBJECT").rstrip().lstrip()
+        if self.translate_imgType(md) != 'flat':
+            return float('nan') # defaults to NaN if not a flat
+        if val[0:3].isdigit():
+            wavelength = float(val[0:3])
+            if wavelength<300: #blindly using the first 3 digits, so if >=1000nm could be bad
```

Comment:
It's probably better to be more precise if the number of digits is not fixed.

It looks like:
```
import itertools
try:
    wavelength = int("".join(itertools.takewhile(str.isdigit,val)))
except ValueError:
    return np.nan
```

or

```
import re
try:
    wavelength = int(re.match('\d+', val).group())
except ValueError:
    return np.nan
```

will work.